apps/experiments/management/commands/migrate_nonpipeline_to_pipeline_experiments.py

```diff
@@ -0,0 +1,214 @@
+        )
+        parser.add_argument(
+            "--experiment-id",
+            type=int,
+            help="Convert only a specific experiment by ID",
+        )
+        parser.add_argument(
+            "--chatbots-flag-only",
+            action="store_true",
+            help='Only convert experiments for teams that have the "flag_chatbots" feature flag enabled',
+        )
+
+    def handle(self, *args, **options):
+        dry_run = options["dry_run"]
+        team_slug = options.get("team_slug")
+        experiment_id = options.get("experiment_id")
+        chatbots_flag_only = options["chatbots_flag_only"]
+
+        query = Q(pipeline__isnull=True) & (Q(assistant__isnull=False) | Q(llm_provider__isnull=False))
+
+        if team_slug:
+            query &= Q(team__slug=team_slug)
+
+        if chatbots_flag_only:
+            chatbots_flag_team_ids = self._get_chatbots_flag_team_ids()
+            if not chatbots_flag_team_ids:
+                self.stdout.write(self.style.WARNING('No teams found with the "flag_chatbots" feature flag enabled.'))
+                return
+            query &= Q(team_id__in=chatbots_flag_team_ids)
+            self.stdout.write(f"Filtering to teams with 'flag_chatbots' FF ({len(chatbots_flag_team_ids)} teams)")
+
+        if experiment_id:
+            query &= Q(id=experiment_id)
+            experiment = Experiment.objects.filter(query).first()
+            if not experiment:
+                self.stdout.write(
+                    self.style.WARNING(f"Experiment {experiment_id} not found or does not need migration.")
+                )
+                return
+
+            if not (experiment.is_default_version or experiment.is_working_version):
+                self.stdout.write(
+                    self.style.WARNING(
+                        f"Experiment {experiment_id} is not a published or unreleased version so does not\
+                        require migration."
+                    )
+                )
+                return
+
+            experiments_to_convert = Experiment.objects.filter(id=experiment.id).select_related(
+                "team", "assistant", "llm_provider", "llm_provider_model"
+            )
+        else:
+            default_experiments = Experiment.objects.filter(query & Q(is_default_version=True))
+            default_working_version_ids = default_experiments.exclude(working_version__isnull=True).values_list(
+                "working_version_id", flat=True
+            )
+            working_experiments = Experiment.objects.filter(query & Q(working_version__isnull=True)).exclude(
+                id__in=default_working_version_ids
+            )
+            combined_ids = list(default_experiments.union(working_experiments).values_list("id", flat=True))
+            experiments_to_convert = Experiment.objects.filter(id__in=combined_ids).select_related(
+                "team", "assistant", "llm_provider", "llm_provider_model"
+            )
```

Comment:
It may be that loading all of these into memory at once could be an issue so it would be prudent to use an iterator:
```suggestion
            experiments_to_convert_queryset = Experiment.objects.filter(id__in=combined_ids).select_related(
                "team", "assistant", "llm_provider", "llm_provider_model"
            )
            # do this here since you can't do a count on an iterator
            experiment_count = experiments_to_convert_queryset.count()
            experiments_to_convert = experiments_to_convert_queryset.iterator(20)
```